uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/payment/BookingPaymentHandler.kt

```diff
@@ -121,6 +122,7 @@ class BookingPaymentHandler @JvmOverloads constructor(
     }
 
     override fun retrieveLoyaltyStatus() {
-        paymentActions?.retrieveLoyaltyStatus()
+        if(LoyaltyFeatureFlags.loyaltyEnabled)
```

Comment:
We should enable or disable the whole feature, not just status retrieval. So at the very beginning there should be a check to see if loyaltyEnabled, where `loyaltyEnabled = !paymentProvider.loyalty.id.isNullOrEmpty && LoyaltyFeatureFlags.loyaltyEnabled`

Reply:
This place where it's now is even before the check for this id in android